src/Menu/menu/sizeModifier.ts

```diff
@@ -0,0 +1,292 @@
+    return [];
+  }
+
+  const oppositePlacement = getOppositePlacement(placement);
+
+  return [
+    getOppositeVariationPlacement(placement),
+    oppositePlacement,
+    getOppositeVariationPlacement(oppositePlacement),
+  ];
+}
+
+interface OverflowDict {
+  placement: Placement;
+  overflow: ReturnType<typeof detectOverflow>;
+  mainVariationSide: BasePlacement;
+  altVariationSide: BasePlacement;
+  basePlacement: ReturnType<typeof getBasePlacement>;
+}
+
+/**
+ * Given a list of `OverflowDict` and the `state`, find the `OverflowDict` that
+ * should be displayed. Returns the first placement that will fit without
+ * scrolling, and if there are none, then the placement that will require the
+ * _leaset_ amount of scrolling.
+ */
+function findMinimumOverflowPlacement({
+  overflowDicts,
+  state,
+}: {
+  overflowDicts: readonly OverflowDict[];
+  state: ModifierArguments<Options>["state"];
+}) {
+  const overflowsWithOverflowMap = overflowDicts.map((overflowDict) => {
+    const { y } = state.modifiersData.preventOverflow || { y: 0 };
+    const { height } = state.rects.popper;
+    const [basePlacement] = overflowDict.placement.split("-");
+
+    const heightProp = basePlacement === "top" ? "top" : "bottom";
+
+    return {
+      overflowDict,
+      overflowPixels: Math.max(0, overflowDict.overflow[heightProp]),
+      // The `overflow` can be negative here; this is how the container can
+      // expand when we are resizing to expose more of the clipped content.
+      maxHeight: height - overflowDict.overflow[heightProp] - y,
+    };
+  });
+
+  overflowsWithOverflowMap.sort((a, b) => {
+    if (b.overflowPixels !== a.overflowPixels) {
+      return a.overflowPixels - b.overflowPixels;
+    }
+
+    // If the heights are the same then we have no overflow in these two
+    // placements. We must prioritize the user-specified placement order here.
+    // While `overflowed` is already in the correct order. Preserving original
+    // sort order is explicitly stated to not be stable in the
+    // [`Array.prototype.sort`
+    // spec](http://www.ecma-international.org/ecma-262/6.0/#sec-array.prototype.sort)
+    //
+    // We could build an index for the indicies, but I don't think we'll have
+    // any performance issues because the placement lists are short.
+    const aPlacementWeight = overflowDicts.findIndex(
+      (overflow) => overflow.placement === a.overflowDict.placement
+    );
+    const bPlacementWeight = overflowDicts.findIndex(
+      (overflow) => overflow.placement === b.overflowDict.placement
+    );
+
+    return aPlacementWeight - bPlacementWeight;
+  });
+
+  return overflowsWithOverflowMap[0];
+}
+
+/**
+ * Padding can be applied to `preventOverflow`, `flip`, or this modifier. Go
+ * through all the modifier options to find it
+ */
+function getPaddingFromState(
+  state: ModifierArguments<Options>["state"]
+): Padding {
+  return state.orderedModifiers.reduce<Padding>((accumulator, modifier) => {
```

Comment:
This seems like a fancy find? `accumulator` made me assume a sum was happening, would result or something be a better name?

Reply:
how do folks feel about 
```ts
state.orderedModifiers.flatMap(
  (modifier) => modifier.options?.padding || []
)[0] || 0;

```

Reply:
This _is_ totally a fancy find. A comment explaining this would help. I need to find the _last_ item in `orderedModifers` that defines `padding` and then default to 0 if one isn't found.